str_analysis/call_non_ref_pathogenic_motifs.py

```diff
@@ -0,0 +1,978 @@
+    """
+
+    if args.genome_version not in ("37", "38"):
+        raise ValueError(f"Unexpected genome version: {args.genome_version}. Must be '37' or '38'")
+
+    chrom, start_1based, end_1based = parse_interval(locus_coords_1based)
+
+    for repeat_unit_number, repeat_unit in enumerate(repeat_units):
+        motif_number = repeat_unit_number + 1 
+        # generate variant catalog
+        variant_catalog_locus_label = f"{locus_id}_{repeat_unit}"
+        offtarget_regions = []
+        if use_offtarget_regions:
+            offtarget_regions = OFFTARGET_REGIONS[args.genome_version][compute_canonical_repeat_unit(repeat_unit)]
+
+        variant_catalog = generate_variant_catalog(
+            variant_catalog_locus_label,
+            repeat_unit, chrom, start_1based, end_1based,
+            offtarget_regions=offtarget_regions)
+
+        variant_catalog_path = f"{variant_catalog_locus_label}.variant_catalog.json"
+        with open(variant_catalog_path, "wt") as f:
+            json.dump([variant_catalog], f)
+
+        # run expansion hunter
+        print("--"*10)
+        print(f"Running ExpansionHunter on {args.sample_id} for repeat unit {repeat_unit}")
+        if args.verbose:
+            print("Using variant catalog: ")
+            pprint(variant_catalog)
+
+        output_prefix = f"{args.sample_id}.{locus_id}_{repeat_unit}.expansion_hunter"
+        expansion_hunter_command = f"""{args.expansion_hunter_path} \
+--sex male \
+--reference "{args.reference_fasta}" \
+--reads "{args.bam_or_cram_path}" \
+--variant-catalog "{variant_catalog_path}" \
+--output-prefix "{output_prefix}" \
+--log-level debug
+"""
+
+        run(expansion_hunter_command, verbose=args.verbose)
+
+        # parse result
+        if os.path.isfile(f"{output_prefix}.json"):
+            with open(f"{output_prefix}.json", "rt") as f:
+                expansion_hunter_output_json = json.load(f)
+        else:
+            print(f"ERROR: ExpansionHunter didn't produce a {output_prefix}.json file. Skipping...")
+            continue
+
+        if args.verbose:
+            print(f"ExpansionHunter output: {pformat(expansion_hunter_output_json)}")
+
+        eh_result = expansion_hunter_output_json.get("LocusResults", {}).get(variant_catalog_locus_label, {}).get(
+            "Variants", {}).get(variant_catalog_locus_label, {})
+
+        if not eh_result:
+            continue
+
+        locus_results_json[f"expansion_hunter_motif{motif_number}_json_output_file"] = f"{output_prefix}.json"
+        locus_results_json[f"expansion_hunter_motif{motif_number}_repeat_unit"] = repeat_unit
+        locus_results_json[f"expansion_hunter_motif{motif_number}_canonical_repeat_unit"] = compute_canonical_repeat_unit(repeat_unit)
+
+        # TODO currently there are no X chromosome loci, but if they do get added, this will be need to be updated
+        # to support single-allele genotypes for males on the X chromosome
```

Comment:
The autosomal genotypes look like  `20/35`  meaning one allele has 20 repeats and the other allele has 35.
For hemizygous genotypes, it'll be just output one number (eg. `35` rather than something like `./35`)